mtags/src/main/scala-3/scala/meta/internal/pc/completions/FilenameCompletions.scala

```diff
@@ -0,0 +1,33 @@
+package scala.meta.internal.pc
+package completions
+
+import scala.meta.internal.mtags.MtagsEnrichments.decoded
+
+import dotty.tools.dotc.ast.tpd.TypeDef
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags
+
+object FilenameCompletions:
+
+  def contribute(
+      filename: String,
+      td: TypeDef
+  )(using ctx: Context): List[CompletionValue] =
+    val owner = td.symbol.owner
+    lazy val scope =
+      owner.info.decls.filter(sym => sym.isType && sym.isDefinedInCurrentRun)
```

Comment:
A minor suggestion. The difference is small but it's enough to check `sourcePos` existence only instead of doing a full check including compiler periods.
```suggestion
      owner.info.decls.filter(sym => sym.isType && sym.sourcePos.exists)
```